cookingWebsite/yourCookbook/viewRecipe.php

```diff
@@ -0,0 +1,112 @@
+<?php
+include('config/init.php');
+verifyUserCooking();
+$recipeId = $_REQUEST['recipeId'];
+$loginId = $_SESSION['loginId'];
+echo yourCookbookEchoHtmlHeader("New Recipe");
+?>
+<script>
+function addTagExistingRecipe(recipeId){
+    var recipeIngredientName = $('#recipeIngredientName').val();
+    var submitItem = $('#submitItem').val();
+    console.log(recipeId, submitItem, recipeIngredientName)
+    $.post('/ajax/saveAddedtag.php', {recipeIngredientName:recipeIngredientName, submitItem:submitItem, recipeId:recipeId}, function(Data){
+          if (Data == "failure"){
+            alert("Error");
+            return false;
+        }
+        // $('.addTag').html(Data)
+    });
+    alert("The tag was sucessfully added!");
+    return false;
+}
+function addRating(loginId, recipeId){
+    var rating = $('#rating').val();
+    var submitItem = $('#submitItem').val();
+    console.log(recipeId, submitItem, loginId)
+    $.post('/ajax/saveRating.php', {rating:rating, submitItem:submitItem, recipeId:recipeId, loginId:loginId}, function(Data){
+          if (Data == "failure"){
+            alert("Error");
+            return false;
+        }
+    });
+    alert("The tag was sucessfully added!");
+    return false;
+}
+function deleteTagExistingRecipe(recipeId){
+    var recipeIngredientName = $('#recipeIngredientName').val();
+    var deleteItem = $('#deleteItem').val();
+    $.post('/ajax/deleteTagExistingPost.php', {recipeIngredientName:recipeIngredientName, deleteItem:deleteItem, recipeId:recipeId}, function(Data){
+          if (Data == "failure"){
+            alert("Error");
+            return false;
+        }
+        // $('.deleteTag').html(Data)
+    });
+    alert("The tag was sucessfully removed!");
+    return false;
+}
+</script>
+<br/><br/><div class = 'recipeHeader'><span class='recipe'>
+
+<?php
+echo recipeContent($recipeId, 'title');
+?>
+ <!-- <a href='#' id='faves' onclick='favorited()'><i class="fa fa-star" aria-hidden="true"></i></a> -->
+</div><p style='text-align:center; font-size:14px'>By:
+<?php
+    echo recipeContent($recipeId, 'firstName')." ".recipeContent($recipeId, 'lastName');
```

Comment:
This function probably isn't the best way to do this. I like that you're thinking about how to use functions to make your code cleaner, but you're calling the db every time you call recipeContent even though the data is the same. There are a couple options here:

1) The most common pattern would be to have a "GetRecipe" function that you call at the top and then reference that each time. Something like 
```
$recipe = getRecipe($recipeId);
echo $recipe['firstName'];
```

2) use a `recipeContent` function like this, but cache the output from the db within that function so that it doesn't re-fetch the data each time. This is more work, but can be useful if you'll be calling this function from all over the place and don't want to worry about whether it's been called before or not. For this case, I'd go with option 1